### PR TITLE
Refactor solver (Part I): Moved power-system data structures out of solve function

### PR DESCRIPTION
**Pull request by dmaldona**
Hi guys,

This is my first stab refactoring the solve function. Here I just have created a new submodule (PowerSystems) with the Pf data structure and a constructor that creates the impedance matrix, voltage, etc...

Since there might be many breaking changes I am doing this refactoring in small parts. Let me know hat you think.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -13,6 +13,7 @@ LineSearches = "d3d80556-e9d4-5f37-9878-2ab0fcc64255"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 Metis = "2679e427-3c69-5b7f-982b-ece356f1e94b"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
+Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
```
Comment (frapac) [inline Project.toml:16]: I would maybe not include `Revise` in `Project.toml`

File: src/ExaPF.jl [hunk 2]
```diff
@@ -36,15 +36,11 @@ include("iterative.jl")
 using .Iterative
 include("network.jl")
 using .Network
-
+include("psystem.jl")
```
Comment (frapac) [inline src/ExaPF.jl:39]: Do you know if we could rename `psystem.jl` `powersystem.jl` (to have more explicit namings)?

File: src/psystem.jl [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+"""
+    Pf
+
+This structure contains constant parameters that define the topology and
+physics of the power network.
+
+# Fields
+- `V::Array{Complex{Float64}}`: voltage
+"""
+struct Pf
```
Comment (frapac) [inline src/psystem.jl:17]: How about renaming `Pf` `PowerFlow`?

File: src/psystem.jl [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+using ..ExaPF: Parse
+using ..ExaPF: Network
+
+using SparseArrays
+
+"""
+    Pf
+
+This structure contains constant parameters that define the topology and
+physics of the power network.
```
Comment (frapac) [inline src/psystem.jl:12]: As far as I understand, `Pf` creates object in the main memory, and not directly onto the GPU. I would maybe add a comment about that here, just to tell the user that some objects are bound to be exported to the GPU after.

File: src/psystem.jl [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    
+        Sbus = assembleSbus(gen, load, SBASE, nbus)
+        
+        new(V, Ybus, data, nbus, ngen, nload, ref, pv, pq, Sbus)
+    end
+end
+
+
+"""
+bustypeindex(data)
```
Comment (frapac) [inline src/psystem.jl:73]: Maybe indent this line for formatting?

File: src/psystem.jl [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    ref = findall(x->x==3, bustype)
+    pv = findall(x->x==2, bustype)
+    pq = findall(x->x==1, bustype)
+
+    return ref, pv, pq
+end
+
+
+"""
+assembleSbus(data)
```
Comment (frapac) [inline src/psystem.jl:119]: I would maybe add an indent as well. Also, it looks like the signature does not match those of the mentioned function.

**Comment by michel2323** [comment]
Maybe we should also put `Parse` and `Network` into the powersystem module.

File: src/ExaPF.jl (hunk 2 above)
Comment (dmaldona) [reply src/ExaPF.jl:39]: Ahh, oversight.

File: src/psystem.jl (hunk 3 above)
Comment (dmaldona) [reply src/psystem.jl:17]: Renamed it to PowerNetwork (referring to the physical object)

File: src/ExaPF.jl (hunk 2 above)
Comment (dmaldona) [reply src/ExaPF.jl:39]: Done

File: Project.toml (hunk 1 above)
Comment (dmaldona) [reply Project.toml:16]: Oversight! It happens too often. Fixed it!

File: src/psystem.jl (hunk 5 above)
Comment (dmaldona) [reply src/psystem.jl:73]: Done

File: src/psystem.jl (hunk 6 above)
Comment (dmaldona) [reply src/psystem.jl:119]: Done

File: src/psystem.jl (hunk 4 above)
Comment (dmaldona) [reply src/psystem.jl:12]: Added comment

**Comment by dmaldona** [comment]
@michel2323 cannot respond directly to your comment. 

Good idea. I have moved `Network` into the powersystem module. The other module, `Parse`, its a bit more trickier to I will create an issue and come back to it later.

**Comment by dmaldona** [comment]
@michel2323 , @frapac : I have pushed a new commit addressing your comments. Please let me know if OK,

**Comment by frapac** [comment]
Looks good to me!

**Pull request merged**